var/spack/repos/builtin/packages/kentutils/package.py

```diff
@@ -7,35 +7,132 @@
 
 
 class Kentutils(MakefilePackage):
-    """Jim Kent command line bioinformatic utilities"""
+    """Jim Kent command line bioinformatic utilities and libraries"""
 
     homepage = "https://genome.cse.ucsc.edu/"
     url = "https://hgdownload.cse.ucsc.edu/admin/exe/userApps.archive/userApps.v453.src.tgz"
 
-    version("459", sha256="0b6e89a183e6385c713cf010a7aeead9da6626d8d2f78c363a4f1bc56ccccebb")
-    # The above archive only goes back to v305. v302 is left for now but deprecated. Suggest
-    # this is dropped on next update (v302 is from 2014!) and the `requires()` removed.
+    maintainers("teaguesterling")
+
+    version("465", sha256="eef17b1f3182d1d9dc99b5c73a6b0468d5d3bd80470f25d3f7706cc1372e04b0")
+    version("464", sha256="24e20fe68e2a2894d802c87662f69a62f71b3c15fafb2e4d6c3c425c63638bb2")
+    version("460", sha256="b955e56ee880074521ef1ab1371491f47e66dc6fdd93b05328386dd675a635fa")
+    version("455", sha256="e458cadad7c4a5c1b8385edafffa1b29380ac725a0c20535bf5a3bab99fe80db")
+    # This version isn't present in the archive any more
+    # Might be worth changing url to: https://github.com/ucscGenomeBrowser/kent-core/tags/...
     version(
-        "302.1",
-        commit="d8376c5d52a161f2267346ed3dc94b5dce74c2f9",
-        git="https://github.com/ENCODE-DCC/kentUtils.git",
+        "459",
+        sha256="0b6e89a183e6385c713cf010a7aeead9da6626d8d2f78c363a4f1bc56ccccebb",
         deprecated=True,
     )
 
-    depends_on("libpng")
-    depends_on("openssl")
-    depends_on("uuid")
-    depends_on("mariadb")
-    depends_on("zlib-api")
-    depends_on("freetype")
-    depends_on("libiconv")
+    variant("libs", default=True, description="Install jk*.a libraries")
+    variant("force_mysql", default=False, description="Force MySQL over MariaDB")
+    variant("htslib", default=False, description="Build and use bundled htslib", sticky=True)
+
+    with default_args(type=("build", "link", "run")):
+        depends_on("libpng")
+        depends_on("openssl")
+        depends_on("uuid")
+        depends_on("mysql-client")
+        depends_on("zlib-api")
+        depends_on("freetype")
+        depends_on("libiconv")
+        depends_on("htslib+pic", when="~htslib+libs")
+
+    # The bgzip.c bug present in other packages is present in kent/src/htslib/bgzf.c
+    # Conflicting line: assert(compressBound(BGZF_BLOCK_SIZE) < BGZF_MAX_BLOCK_SIZE);
+    # We can patch this by removing the assertion, but there are still performance issues
+    # See: https://github.com/samtools/htslib/issues/1257
+    conflicts("zlib-ng")
+
+    # Does not add a link to mysql_config, which is required for compilation
+    conflicts("mariadb-c-client")
 
-    requires("%gcc", when="@302.1")
+    # MariaDB can take a very long time to compile if you just need the c client
+    conflicts("mariadb", when="+force_mysql")
```

Comment:
Would it be better to leave this out of the variants and let users handle it in the spec?